### ISSUE TITLE
Docs for persistent compilation cache are superseded

### ISSUE THREAD

**Issue by epignatelli** [google/jax#21908]
### Description

I followed the docs here: https://jax.readthedocs.io/en/latest/persistent_compilation_cache.html, but I still do not get cache hits. 

Is there something I am missing?

```python
import os
import jax

cache_dir = "/tmp/jax_cache/"
os.makedirs(cache_dir, exist_ok=True)
jax.config.update("jax_compilation_cache_dir", cache_dir)


def f(x, y):
    return jax.numpy.matmul(x, y)

x = jax.numpy.ones((2, 2))
jax.jit(f)(x, x.T)


!ls -a /tmp/jax_cache/
```

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.4.26
jaxlib: 0.4.26
numpy:  1.25.2
python: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
jax.devices (1 total, 1 local): [CpuDevice(id=0)]
process_count: 1
platform: uname_result(system='Linux', node='1210851b0b9c', release='6.1.85+', version='#1 SMP PREEMPT_DYNAMIC Sun Apr 28 14:29:16 UTC 2024', machine='x86_64')
```

**Comment by ayaka14732** [comment]
The flags are confusing. You need to do:

```python
jax.config.update('jax_compilation_cache_dir', '/tmp/jax_cache')
jax.config.update('jax_persistent_cache_min_entry_size_bytes', 1)
jax.config.update('jax_persistent_cache_min_compile_time_secs', 0.0000000000001)
```

You may also be interested in https://github.com/google/jax/pull/21394

**Comment by superbobry** [comment]
I think we should update the docs to explain all the relevant flags, now that we also have LRU eviction.

**Comment by epignatelli** [comment]
Great, thanks both for the heads-up, I'll rename the issue to doc updates. 
I cannot update the label, unfortunately.

**Comment by ayaka14732** [comment]
@epignatelli Do you want to send a PR to update the doc?

**Comment by Bhavay-2001** [comment]
Hi @ayaka14732, I haven't worked on `JAX` and this would be my first issue in this repo but if it is something that I can solve, I would love to open the PR.

**Comment by ayaka14732** [comment]
@Bhavay-2001 Thank you for the interest, but let's see if @epignatelli would like to open the PR first.

**Comment by Bhavay-2001** [comment]
Alright @ayaka14732. Any other minor issue from which I can start working with `jax`. I will start to learn and implement it in the coming days and then proceed with more advanced issues. Thanks

**Comment by ayaka14732** [comment]
@Bhavay-2001 Great question! You can open a discussion in https://github.com/google/jax/discussions and maybe other maintainers can give you suggestions

**Comment by ayaka14732** [comment]
Seems that @keshavb96 is working on this. Let's fix this on their side.

**Comment by mattjj** [comment]
@ayaka14732 since #21819 is merged, can we close this issue?

**Comment by ayaka14732** [comment]
Forgot this! Thanks!
